lib/scsi_defs.h

```diff
@@ -803,6 +811,56 @@ struct UnmapBlockDescriptor {
 } ABSL_ATTRIBUTE_PACKED;
 static_assert(sizeof(UnmapBlockDescriptor) == 16);
 
+// SCSI Reference Manual Table 483
+// https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+struct SupportedVitalProductData {
+  PeripheralQualifier peripheral_qualifier : 3;
+  PeripheralDeviceType peripheral_device_type : 5;
+  uint8_t page_code : 8;
+  uint8_t _reserved : 8;
+  uint8_t page_length : 8;
+  uint8_t supported_page_list[256];
+} ABSL_ATTRIBUTE_PACKED;
+static_assert(sizeof(SupportedVitalProductData) == 260);
+
+// SCSI Reference Manual Table 484
+// https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+struct UnitSerialNumber {
+  PeripheralQualifier peripheral_qualifier : 3;
+  PeripheralDeviceType peripheral_device_type : 5;
+  uint8_t page_code : 8;
+  uint8_t _reserved : 8;
+  uint8_t page_length : 8;
+  uint8_t product_serial_number[256];
+} ABSL_ATTRIBUTE_PACKED;
+static_assert(sizeof(UnitSerialNumber) == 260);
+
+// SCSI Reference Manual Table 460
+// https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+struct IdentificationDescriptor {
+  uint8_t protocol_identifier : 4;
+  uint8_t code_set : 4;
+  uint8_t piv : 1;
+  uint8_t _reserved1 : 1;
+  uint8_t association : 2;
+  uint8_t identifier_type : 4;
+  uint8_t _reserved2 : 8;
+  uint8_t identifier_length : 8;
+  uint8_t identifier[256];
+} ABSL_ATTRIBUTE_PACKED;
+static_assert(sizeof(IdentificationDescriptor) == 260);
+
+// SCSI Reference Manual Table 459
+// https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+struct DeviceIdentificationVpd {
+  PeripheralQualifier peripheral_qualifier : 3;
+  PeripheralDeviceType peripheral_device_type : 5;
+  uint8_t page_code : 8;
+  uint8_t page_length : 8;
+  IdentificationDescriptor identification_descriptor_list[256];
```

Comment:
Same here - we should not allocate that many structs.

Reply:
what should we do if the received command has the max page_length?

Reply:
I think @harloff-g and/or @basimsahaf worked on some SCSI structs that were variable sized. I think in previous cases we left them out of the struct definition and would deal with them another way

Reply:
@w-michal I've removed variable length lists from scsi response objects to be handled during writing to the scsi response buffer. This is consistent with how we're defining other variable scsi command structs.